build_tools/analyze_memory_logs.py

```diff
@@ -0,0 +1,336 @@
+                lines.append(f"  Memory Usage:")
+                lines.append(f"    Average: {analysis['avg_memory_percent']:.1f}%")
+                lines.append(
+                    f"    Peak: {analysis['max_memory_percent']:.1f}% ({analysis['peak_memory_gb']:.2f} GB)"
+                )
+                lines.append(
+                    f"    Range: {analysis['min_memory_percent']:.1f}% - {analysis['max_memory_percent']:.1f}%"
+                )
+                lines.append(f"  Swap Usage:")
+                lines.append(f"    Average: {analysis['avg_swap_percent']:.1f}%")
+                lines.append(f"    Peak: {analysis['max_swap_percent']:.1f}%")
+                lines.append(f"  Time Range:")
+                lines.append(f"    Start: {analysis['start_time']}")
+                lines.append(f"    End: {analysis['end_time']}")
+
+                if analysis["severity"] in ["CRITICAL", "HIGH"]:
+                    lines.append(
+                        f"  [!] WARNING: This phase had {analysis['severity']} memory usage!"
+                    )
+
+        # Key findings
+        lines.append("")
+        lines.append("KEY FINDINGS")
+        lines.append("=" * 80)
+
+        critical_phases = [
+            a for a in phase_analyses if a["severity"] in ["CRITICAL", "HIGH"]
+        ]
+        if critical_phases:
+            lines.append("")
+            lines.append("[!] PHASES WITH HIGH MEMORY USAGE:")
+            for analysis in critical_phases:
+                lines.append(
+                    f"  - {analysis['phase']}: {analysis['max_memory_percent']:.1f}% peak"
+                )
+                lines.append(
+                    f"    Consider reducing parallel jobs or increasing available memory"
+                )
+
+        high_swap = [a for a in phase_analyses if a["max_swap_percent"] > 50]
+        if high_swap:
+            lines.append("")
+            lines.append("[!] PHASES WITH HIGH SWAP USAGE:")
+            for analysis in high_swap:
+                lines.append(
+                    f"  - {analysis['phase']}: {analysis['max_swap_percent']:.1f}% peak"
+                )
+                lines.append(
+                    f"    Swap usage indicates memory pressure and will slow down builds"
+                )
+
+        lines.append("")
+        lines.append("=" * 80)
+
+        return "\n".join(lines)
+
+    def write_github_summary(self, phase_analyses: List[Dict[str, Any]]):
+        """Write analysis to GitHub Actions step summary."""
+        if "GITHUB_STEP_SUMMARY" not in os.environ:
+            return
+
+        try:
+            with open(os.environ["GITHUB_STEP_SUMMARY"], "a") as f:
+                f.write("\n## Memory Usage Analysis\n\n")
+                f.write("### Peak Memory Usage by Phase\n\n")
+                f.write("| Phase | Peak Memory | Avg Memory | Peak Swap | Severity |\n")
+                f.write("|-------|-------------|------------|-----------|----------|\n")
```

Comment:
Use https://github.com/ROCm/TheRock/blob/main/build_tools/github_actions/github_actions_utils.py for writing to `GITHUB_STEP_SUMMARY`